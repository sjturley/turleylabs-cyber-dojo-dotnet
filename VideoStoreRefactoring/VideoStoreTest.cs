using System;

using NUnit.Framework;

using ApprovalTests;
using ApprovalTests.Core;
using ApprovalTests.Reporters;
using ApprovalUtilities;

[TestFixture]
[UseReporter(typeof(QuietReporter))]
public class VideoStoreTest
{
    [OneTimeSetUp]
    public void OneTimeSetup()
    {
        Approvals.RegisterDefaultNamerCreation(() => new CyberDojoNamer());
    }
    
    [Test]
    public void RENAME_ME()
    {
        Approvals.Verify("Me");
    }
}
