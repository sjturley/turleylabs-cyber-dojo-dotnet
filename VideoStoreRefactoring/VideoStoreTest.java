import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.QuietReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;

@UseReporter(QuietReporter.class)
public class VideoStoreTest {
    @Test
    public void approveCustomerStatement() {
        CombinationApprovals.verifyAllCombinations(this::createCustomerStatement
                , new Integer[] {Movie.NEW_RELEASE, Movie.REGULAR, Movie.CHILDRENS}
                , new Integer[] {1, 2, 3, 4});
    }

    private String createCustomerStatement(int priceCode, int daysRented) {
        Customer customer = new Customer("Steve");
        Movie movie = new Movie("Star Wars", Movie.REGULAR);
        customer.addRental (new Rental(movie, daysRented));
        movie.setPriceCode(priceCode);
        return customer.statement();
    }
}
