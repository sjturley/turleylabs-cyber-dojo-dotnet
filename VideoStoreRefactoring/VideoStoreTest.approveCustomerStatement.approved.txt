[1, 1] => Rental Record for Steve
	Star Wars	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
[1, 2] => Rental Record for Steve
	Star Wars	6.0
Amount owed is 6.0
You earned 2 frequent renter points 
[1, 3] => Rental Record for Steve
	Star Wars	9.0
Amount owed is 9.0
You earned 2 frequent renter points 
[1, 4] => Rental Record for Steve
	Star Wars	12.0
Amount owed is 12.0
You earned 2 frequent renter points 
[0, 1] => Rental Record for Steve
	Star Wars	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[0, 2] => Rental Record for Steve
	Star Wars	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[0, 3] => Rental Record for Steve
	Star Wars	3.5
Amount owed is 3.5
You earned 1 frequent renter points 
[0, 4] => Rental Record for Steve
	Star Wars	5.0
Amount owed is 5.0
You earned 1 frequent renter points 
[2, 1] => Rental Record for Steve
	Star Wars	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[2, 2] => Rental Record for Steve
	Star Wars	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[2, 3] => Rental Record for Steve
	Star Wars	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[2, 4] => Rental Record for Steve
	Star Wars	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
